crux-core/src/crux/tx.clj

```diff
@@ -233,6 +234,16 @@
 (defmethod index-tx-event :crux.tx/delete [[op k start-valid-time end-valid-time] tx deps]
   {:etxs (put-delete-etxs k start-valid-time end-valid-time nil tx deps)})
 
+(defmethod index-tx-event :crux.tx/match [[op k v at-valid-time :as match-op]
+                                          {:crux.tx/keys [tx-time tx-id] :as tx}
+                                          {:keys [history] :as deps}]
+  {:pre-commit-fn #(let [{:keys [content-hash] :as entity} (entity-at history
```

Comment:
Feels like it make sense to share the logic between CAS and match here? That is, CAS delegates to the same logic match uses. My impression is that all CAS tests have been removed (or rather, converted), so feels better to have indirect coverage and same behaviour for both.

Reply:
mm. :thinking: I didn't because of the additional object store check in the original CaS (needed because of the sorted-maps rehash legacy issue), which isn't required in the new match, and also causes other issues, like #528. not sure though?

Reply:
looked over it this morning - we've still got the test-can-cas-entity (amongst others) in tx-test. if/when we remove CaS as a user-visible tx-op, we can use a similar pattern to when we removed the evict time ranges - creating tests that manually inserted legacy events in the tx-log.